### PR TITLE
ensure progress tracker completes execution

### PR DESCRIPTION
**Pull request by jonasagx**
Fixes [#764](https://github.com/anchore/syft/issues/764)
Signed-off-by: Jonas Galvão Xavier <jonas.agx@gmail.com>

### CODE REVIEW

File: pkg/image/docker/daemon_provider.go [hunk 1]
```diff
@@ -193,7 +205,7 @@ func (p *DaemonImageProvider) Provide() (*image.Image, error) {
 		return nil, fmt.Errorf("unable to save image to tar: %w", err)
 	}
 	if nBytes == 0 {
-		return nil, fmt.Errorf("cannot provide an empty image")
+		return nil, ErrEmptyImage
```
Comment (kzantow) [inline pkg/image/docker/daemon_provider.go:208]: is there a specific reason the `ErrEmptyImage` var was introduced?

**Review by kzantow — APPROVED** [review]
This seems good to me, not being especially familiar with why the `copyProgress.SetComplete()` wasn't being called before

File: pkg/image/docker/daemon_provider.go [hunk 2]
```diff
@@ -182,9 +197,6 @@ func (p *DaemonImageProvider) Provide() (*image.Image, error) {
 		}
 	}()
 
-	// cancel indeterminate progress
-	estimateSaveProgress.SetCompleted()
```
Comment (wagoodman) [inline pkg/image/docker/daemon_provider.go:186]: this should probably stick around. Why? The image save progress is only a guess (a timer counting up to a particular time where the overall progress would be considered at 50%). It's logical to adjust the first image save timer to complete when the image save operation returns.

File: pkg/image/docker/daemon_provider.go (hunk 1 above)
Comment (jonasagx) [reply pkg/image/docker/daemon_provider.go:208]: I thought it would be useful to have that error as a var, but no one references that, I will drop it

**Pull request merged**